cmd/provisioner-localpv/app/config.go

```diff
@@ -37,12 +37,45 @@ const (
 	//KeyPVStorageType defines if the PV should be backed
 	// a hostpath ( sub directory or a storage device)
 	KeyPVStorageType = "StorageType"
+
 	//KeyPVBasePath defines base directory for hostpath volumes
 	// can be configured via the StorageClass annotations.
 	KeyPVBasePath = "BasePath"
+
 	//KeyPVFSType defines filesystem type to be used with devices
 	// and can be configured via the StorageClass annotations.
 	KeyPVFSType = "FSType"
+
+	//KeyBDPoolName defines the value for the Block Device Pool
+	//label selector configured via the StorageClass annotations.
+	//User can group block devices across nodes to form a block
+	//device pool, by setting the label on block devices as:
+	//  openebs.io/block-device-pool=<pool-name>
+	//
+	//The <pool-name> used above can be passsed to the
+	//Local PV device provisioner via the StorageClass
+	//CAS annotations.
+	//
+	//Example: Local PV device StorageClass for picking devices
+	//labeled as: openebs.io/block-device-pool=pool-x
+	//will be as follows
+	//
+	// kind: StorageClass
+	// metadata:
+	//   name: openebs-device-pool-x
+	//   annotations:
+	//     openebs.io/cas-type: local
+	//     cas.openebs.io/config: |
+	//       - name: StorageType
+	//         value: "device"
+	//       - name: BlockDevicePoolName
+	//         value: "pool-x"
+	// provisioner: openebs.io/local
+	// volumeBindingMode: WaitForFirstConsumer
+	// reclaimPolicy: Delete
+	//
+	KeyBDPoolName = "BlockDevicePoolName"
```

Comment:
can this be called as 'LocalPVPoolName'?
Not sure how BlockDevicePoolName makes sense here

Reply:
The name is being derived from label  `openebs.io/block-device-pool`, is set on the block devices. 

Just thinking out loud, the `pool` concept can be implemented using this label selector. We need a fixed label key that is shared by NDM and all its consumers. "Pool" may be a over-loaded term. 

How about using "openebs.io/block-device-tag"? And make the corresponding naming changes to the above Key, and other method names and variables.

Reply:
Is this related to NDM? If yes then label key should have some indication.
I could think of below label key

`blockdevice.ndm.openebs.io/tag`

Reply:
If we want to go with pool then `blockdevice.ndm.openebs.io/pool-name`

Reply:
If this label is applicable to BlockDevice & BlockDeviceClaim then NDM is not required in the label key. Sorry for above confusions.

So we can have `openebs.io/block-device-tag`. The one you had suggested initially.

Reply:
with this stmt, `We need a fixed label key that is shared by NDM and all its consumers` are we saying - every NDM consumer should be fixed to this label? We don't need this as there is 'Selector' in spec of BD. If it was part of some labels of BD, it might be the case.
By fixing it to fixed string, 'Selector' in 'spec' of BD can be removed and moved to label of 'BD'.
If this is just for local pv, this key has to get into StorageClass, and in the 'Selector' of BD.
Is my understanding correct?

Reply:
it will be understood by NDM and Local PV (and other storage engines possibly in future):
- NDM will have special handling for this label. Like if BD has label, provide it to only those BDCs coming with the label. 
- Local PV will use it by reading it from Storage Class and passing it to BDC
- CSPC operator might label the block devices with a tag, before claiming them.

Reply:
is this the same tag for every consumer of NDM?

Reply:
can we get example of BD for above example?

Reply:
I meant - example mentioned in PR